bin/portal-bridge/src/bridge/e2hs.rs

```diff
@@ -0,0 +1,439 @@
+use std::{
+    str::FromStr,
+    sync::{Arc, Mutex},
+};
+
+use alloy::primitives::B256;
+use anyhow::{anyhow, ensure};
+use clap::Parser;
+use e2store::e2hs::{BlockTuple, E2HS};
+use ethportal_api::{
+    jsonrpsee::http_client::HttpClient,
+    types::execution::{
+        block_body::BlockBody, header_with_proof::HeaderWithProof, receipts::Receipts,
+    },
+    HistoryContentKey, HistoryContentValue,
+};
+use futures::future::join_all;
+use rand::seq::SliceRandom;
+use tokio::{
+    sync::{OwnedSemaphorePermit, Semaphore},
+    task::JoinHandle,
+    time::{sleep, timeout, Duration},
+};
+use tracing::{debug, error, info, warn};
+use trin_metrics::bridge::BridgeMetricsReporter;
+use trin_validation::header_validator::HeaderValidator;
+
+use crate::{
+    bridge::history::{HEADER_SATURATION_DELAY, SERVE_BLOCK_TIMEOUT},
+    put_content::gossip_history_content,
+    stats::{HistoryBlockStats, StatsReporter},
+    types::range::block_range_to_epochs,
+};
+
+/// Looks for the e2hs file for the given epoch index in a local directory.
+/// TODO: this will need to be updated to fetch files from a hosted
+/// server once we have a proper source for the files.
+fn get_e2hs_file(epoch_index: u64) -> anyhow::Result<String> {
+    let e2hs_dir = "./test-e2hs";
+    let all_files = std::fs::read_dir(e2hs_dir)?;
+    let files: Vec<String> = all_files
+        .filter_map(|entry| {
+            let entry = entry.ok()?;
+            let path = entry.path();
+            let path_str = path.to_str()?;
+            if path_str.contains(&format!("mainnet-{epoch_index:05}")) {
+                Some(path_str.to_string())
+            } else {
+                None
+            }
+        })
+        .collect();
+    let file_count = files.len();
+    ensure!(
+        file_count == 1,
+        "Expected 1 file for epoch {epoch_index}, found {file_count}"
+    );
+    let file = files.first().expect("to be able to get first file");
+    Ok(file.clone())
+}
```

Comment:
```suggestion
fn get_e2hs_file(epoch_index: u64) -> anyhow::Result<String> {
    let e2hs_dir = "./test-e2hs";
    let all_files: Vec<String> = std::fs::read_dir(e2hs_dir)?
        .filter_map(Result::ok)
        .filter_map(|entry| entry.path().to_str().map(str::to_string))
        .collect();
    let e2hs_path = all_files
        .into_iter()
        .find(|path| path.contains(&format!("mainnet-{epoch_index:05}-")))
        .ok_or_else(|| anyhow::anyhow!("Failed to find e2hs file for epoch {epoch_index}"))?;
    Ok(e2hs_path)
}
```

What do you think about doing it like this?